examples/navigation/epics/fetchReposByUser.js

```diff
@@ -0,0 +1,36 @@
+import { merge } from 'rxjs/observable/merge';
+import { LOCATION_CHANGE } from 'react-router-redux';
+import { ajax } from 'rxjs/observable/dom/ajax';
+import * as ActionTypes from '../actionTypes';
+
+// Hack since react-router-redux doesn't hook
+// into react-router to grab parsed params.
+const USER_PATTERN = /repos\/([a-zA-Z0-9_]+)/;
+
+export default function fetchReposByUser(action$) {
+  const fetchIntents$ = action$.ofType(LOCATION_CHANGE)
+    .map(
+      ({ payload }) => payload.pathname.match(USER_PATTERN)
+    )
+    .filter(match => !!match)
+    .map(([, user]) => user);
+  return merge(
+    fetchIntents$.map(user => ({
+      type: ActionTypes.REQUESTED_USER_REPOS,
+      payload: {
+        user
+      }
+    })),
+    fetchIntents$
+      .switchMap(user =>
+        ajax.getJSON(`https://api.github.com/users/${user}/repos`)
+          .map(res => ({
+            type: ActionTypes.RECEIVED_USER_REPOS,
+            payload: {
+              repos: res,
+              user
+            }
+          }))
```

Comment:
>  if they put a .catch() that returns an error action inside the switchMap, now will wrongfully be mapped as the success response

I can only assume you mean they're trying to handle some error that could have occurred during the mapping of the response? If so, then you're right.

I'm mostly indifferent, but *I personally* would like people to know about the projection function on `switchMap`, `flatMap`, `concatMap`, etc.

Then again, when/if I ever get the sync operations flattening to land, it shouldn't matter as much. There will still be an added closure, which will slow things down, but it will work.

I retract my suggestion. Mostly because Jay cares more than I do. haha.